packages/compass-aggregations/src/utils/get-schema.ts

```diff
@@ -0,0 +1,40 @@
+import type { Document } from 'mongodb';
+
+const getArrayKeys = (records: Document[]) => {
+  return records
+    .map((item) => getObjectKeys(item))
+    .flat()
+    .filter((x, i, a) => a.indexOf(x) === i)
```

Comment:
```suggestion
```
Now we don't need this I guess, right?

Reply:
we do need it here. here we do flat on the list of keys and then it can have duplicates. so we are filtering them here.

Reply:
ahh that's right